internal/cli/csrconfig.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/spf13/pflag"
+)
+
+type CSRConfigOptions struct {
+	GlobalOptions
+	CertSigningRequestFile string
+	Outfile                string
+	Name                   string
+	ExpirationSeconds      string
+}
+
+func DefaultCSRConfigOptions() *CSRConfigOptions {
+	return &CSRConfigOptions{
+		GlobalOptions:          DefaultGlobalOptions(),
+		CertSigningRequestFile: "",
+		Outfile:                "",
+		Name:                   "",
+		ExpirationSeconds:      "",
+	}
+}
+
+func NewCmdCSRConfig() *cobra.Command {
+	o := DefaultCSRConfigOptions()
+	cmd := &cobra.Command{
+		Use:   "csr-generate",
+		Short: "generate a CSR resource config .yaml based on a CSR file .csr",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *CSRConfigOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Outfile, "outfile", "o", "mygeneratedcsr.yaml", "specify a filename in which to save the generated csr resource config")
+	fs.StringVarP(&o.Name, "name", "n", "mycsr", "specify a name for the csr")
+	fs.StringVarP(&o.ExpirationSeconds, "expiry", "e", "604800", "specify desired certificate expiration in seconds")
+}
+
+func (o *CSRConfigOptions) Complete(cmd *cobra.Command, args []string) error {
+	if err := o.GlobalOptions.Complete(cmd, args); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *CSRConfigOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	_, err := os.ReadFile(args[0])
+	if err != nil {
+		return fmt.Errorf("invalid CSR file specified")
+	}
+
+	return nil
+}
+
+func (o *CSRConfigOptions) Run(ctx context.Context, args []string) error {
+	data, err := os.ReadFile(args[0])
+	if err != nil {
+		return fmt.Errorf("invalid CSR file specified")
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(data)
+
+	output1 := `apiVersion: v1alpha1
+kind: CertificateSigningRequest
+metadata:
+  name: `
+	output2 := `
+spec:
+  request: `
+
+	output3 := `
+  signerName: ca
+  usages: ["clientAuth", "CA:false"]
+  expirationSeconds: `
+	filename := createFilename(o.Outfile)
+
+	f, err := os.Create(filename)
+	if err != nil {
+		return err
+	}
+
+	expiry := fmt.Sprintf(o.ExpirationSeconds + "\n")
+	output := fmt.Sprintf(output1 + o.Name + output2 + encoded + output3 + expiry)
+	_, err = f.WriteString(output)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("config file written to: %s\n", filename)
+
+	return nil
+}
+
+func createFilename(filename string) string {
+	fileroot := filename
+	if filename[len(filename)-5:] == ".yaml" {
+		fileroot = filename[:len(filename)-5]
+	}
+
+	// if file already exists, don't overwrite it; add suffix
+	if _, err := os.Stat(filename); err == nil {
```

Comment:
Check error type?

Reply:
My understanding is that in this case any error will indicate that the file does not exist, so we can safety write a file with that name. But let me know if this is incorrect.